src/ADNLPProblems/AMPGO13.jl

```diff
@@ -3,7 +3,7 @@ export AMPGO13
 function AMPGO13(; n::Int = default_nvar, type::Val{T} = Val(Float64), kwargs...) where {T}
   function f(x)
     n = length(x)
-    return 0 < x[1] < 1 ? -T(x[1]^(2 / 3) + (1 - x[1]^2)^(1 / 3)) : T[Inf]
+    return 0 < x[1] < 1 ? -(x[1]^T(2 / 3) + (1 - x[1]^2)^T(1 / 3)) : T[Inf]
```

Comment:
I don't think it's necessary to enclose Ints or rationals with `T` either. Also `T[Inf]` could again cause problem. Is there some sort of `convert(eltype(x), Inf)`?

Reply:
Also `T(1 / 3)` will create Float64 1/3 and then convert to `T`, which for `BigFloat` is loss of precisions. `(T(2) / 3)` is uglier but better.

Reply:
I think we're supposed to leave it  as `2 / 3` to be generic.

Reply:
What do you mean by generic?

Reply:
I mean that Ints and rationals will be converted automatically to the appropriate floating-point type.

Reply:
`2 / 3` is just the `Int64` division. We can use `2 // 3` instead to have Rational, but that might be slower.

Reply:
No actually 2/3 is a Float64. Let's use 2//3 then.